<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Quadro</class>
 <widget class="QMainWindow" name="Quadro">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1054</width>
    <height>566</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Quadro</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="formLayoutWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>450</y>
      <width>271</width>
      <height>58</height>
     </rect>
    </property>
    <layout class="QFormLayout" name="ports">
     <property name="fieldGrowthPolicy">
      <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
     </property>
     <item row="0" column="0">
      <widget class="QLabel" name="label_device">
       <property name="text">
        <string>Quadro port</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Joystick port</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QComboBox" name="joystick_device">
       <property name="editable">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QComboBox" name="quadro_device">
       <property name="editable">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QTabWidget" name="tabWidget_mode">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>421</width>
      <height>121</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab_manual">
     <attribute name="title">
      <string>Manual</string>
     </attribute>
     <widget class="QWidget" name="formLayoutWidget_5">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>414</width>
        <height>81</height>
       </rect>
      </property>
      <layout class="QFormLayout" name="formLayout_5">
       <item row="0" column="0">
        <widget class="QLabel" name="label_power">
         <property name="text">
          <string>Force</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSlider" name="power">
         <property name="maximum">
          <number>10</number>
         </property>
         <property name="singleStep">
          <number>1</number>
         </property>
         <property name="tracking">
          <bool>true</bool>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TicksBothSides</enum>
         </property>
         <property name="tickInterval">
          <number>1</number>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_torque_manual_correction">
         <property name="text">
          <string>Torque manual</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="sizeConstraint">
          <enum>QLayout::SetMinimumSize</enum>
         </property>
         <item>
          <widget class="QLabel" name="label_torque_manual_correction_x">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>x</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="torque_manual_correction_x">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="suffix">
            <string/>
           </property>
           <property name="decimals">
            <number>5</number>
           </property>
           <property name="minimum">
            <double>-1.000000000000000</double>
           </property>
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.030000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Fixed</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>30</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="label_torque_manual_correction_y">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>y</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="torque_manual_correction_y">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="suffix">
            <string/>
           </property>
           <property name="decimals">
            <number>5</number>
           </property>
           <property name="minimum">
            <double>-1.000000000000000</double>
           </property>
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.030000000000000</double>
           </property>
           <property name="value">
            <double>0.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_joystick">
     <attribute name="title">
      <string>Joystick controlled</string>
     </attribute>
     <widget class="QWidget" name="formLayoutWidget_4">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>411</width>
        <height>91</height>
       </rect>
      </property>
      <layout class="QFormLayout" name="formLayout_4">
       <property name="fieldGrowthPolicy">
        <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="label_joystick_data">
         <property name="text">
          <string>J Data</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="joystick_data">
         <property name="font">
          <font>
           <family>Terminus</family>
           <pointsize>12</pointsize>
          </font>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="2">
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QDial" name="joystick_power">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="maximum">
            <number>100</number>
           </property>
           <property name="singleStep">
            <number>0</number>
           </property>
           <property name="tracking">
            <bool>false</bool>
           </property>
           <property name="wrapping">
            <bool>false</bool>
           </property>
           <property name="notchTarget">
            <double>29.000000000000000</double>
           </property>
           <property name="notchesVisible">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
   <widget class="QWidget" name="formLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>124</y>
      <width>428</width>
      <height>301</height>
     </rect>
    </property>
    <layout class="QFormLayout" name="formLayout_2">
     <property name="fieldGrowthPolicy">
      <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
     </property>
     <item row="2" column="0">
      <widget class="QLabel" name="label_readwrite_time">
       <property name="text">
        <string>Read/Write/Loop</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLabel" name="readwrite_time">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_throttle">
       <property name="text">
        <string>Torque</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLineEdit" name="torque">
       <property name="font">
        <font>
         <family>Terminus</family>
         <pointsize>12</pointsize>
        </font>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label_gyro">
       <property name="text">
        <string>Gyro data</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QLineEdit" name="gyro">
       <property name="font">
        <font>
         <family>Terminus</family>
         <pointsize>12</pointsize>
        </font>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="label_accel">
       <property name="text">
        <string>Accel. data</string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QLineEdit" name="accel">
       <property name="font">
        <font>
         <family>Terminus</family>
         <pointsize>12</pointsize>
        </font>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="label_angle">
       <property name="text">
        <string>Angle (filtered)</string>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="QLineEdit" name="angle">
       <property name="font">
        <font>
         <family>Terminus</family>
         <pointsize>12</pointsize>
        </font>
       </property>
      </widget>
     </item>
     <item row="7" column="0">
      <widget class="QLabel" name="label_motors">
       <property name="text">
        <string>Motors power</string>
       </property>
      </widget>
     </item>
     <item row="7" column="1">
      <widget class="QLineEdit" name="motors">
       <property name="font">
        <font>
         <family>Terminus</family>
         <pointsize>12</pointsize>
        </font>
       </property>
      </widget>
     </item>
     <item row="8" column="0">
      <widget class="QLabel" name="label_reaction_type">
       <property name="text">
        <string>Reaction type</string>
       </property>
      </widget>
     </item>
     <item row="8" column="1">
      <widget class="QComboBox" name="reaction_type">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <item>
        <property name="text">
         <string>None</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Angular velocity PID</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Acceleration (doesn't work for now)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Angle PID</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="9" column="0">
      <widget class="QLabel" name="label_PID_angle">
       <property name="text">
        <string>PID angle</string>
       </property>
      </widget>
     </item>
     <item row="9" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="sizeConstraint">
        <enum>QLayout::SetMinimumSize</enum>
       </property>
       <item>
        <widget class="QLabel" name="PID_angle_Kp_label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Kp</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="PID_angle_Kp">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="suffix">
          <string/>
         </property>
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="minimum">
          <double>-10.000000000000000</double>
         </property>
         <property name="maximum">
          <double>10.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
         <property name="value">
          <double>0.000000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="PID_angle_Ki_label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Ki</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="PID_angle_Ki">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="suffix">
          <string/>
         </property>
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="minimum">
          <double>-10.000000000000000</double>
         </property>
         <property name="maximum">
          <double>10.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="PID_angle_Kd_label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Kd</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="PID_angle_Kd">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="suffix">
          <string/>
         </property>
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="minimum">
          <double>-1.000000000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
         <property name="value">
          <double>0.000000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="10" column="0">
      <widget class="QLabel" name="label_PID_angular_velocity">
       <property name="text">
        <string>PID angle'</string>
       </property>
      </widget>
     </item>
     <item row="10" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <property name="sizeConstraint">
        <enum>QLayout::SetMinimumSize</enum>
       </property>
       <item>
        <widget class="QLabel" name="PID_angular_velocity_Kp_label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Kp</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="PID_angular_velocity_Kp">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="suffix">
          <string/>
         </property>
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="minimum">
          <double>-10.000000000000000</double>
         </property>
         <property name="maximum">
          <double>10.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
         <property name="value">
          <double>0.000000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="PID_angular_velocity_Ki_label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Ki</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="PID_angular_velocity_Ki">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="suffix">
          <string/>
         </property>
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="minimum">
          <double>-10.000000000000000</double>
         </property>
         <property name="maximum">
          <double>10.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="PID_angular_velocity_Kd_label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Kd</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="PID_angular_velocity_Kd">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="suffix">
          <string/>
         </property>
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="minimum">
          <double>-1.000000000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_voltage">
       <property name="text">
        <string>Voltage</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QProgressBar" name="voltage">
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
       <property name="format">
        <string/>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QTabWidget" name="tabwidgetplot">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>4</y>
      <width>621</width>
      <height>501</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tabgyro_data">
     <attribute name="title">
      <string>Gyroscope data</string>
     </attribute>
     <widget class="QwtPlot" name="plot_gyro">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>617</width>
        <height>471</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tabacc_data">
     <attribute name="title">
      <string>Accelerometer data</string>
     </attribute>
     <widget class="QwtPlot" name="plot_acc">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>617</width>
        <height>471</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tabangle">
     <attribute name="title">
      <string>Angle</string>
     </attribute>
     <widget class="QwtPlot" name="plot_angle">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>617</width>
        <height>471</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tabthrottle">
     <attribute name="title">
      <string>Torques and force</string>
     </attribute>
     <widget class="QwtPlot" name="plot_torques_and_force">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>617</width>
        <height>471</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tabcorrections">
     <attribute name="title">
      <string>Corrections</string>
     </attribute>
     <widget class="QwtPlot" name="plot_corrections">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>617</width>
        <height>471</height>
       </rect>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1054</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuQuadro">
    <property name="title">
     <string>Quadro</string>
    </property>
    <addaction name="actionQuadroConnect"/>
    <addaction name="actionQuadroDisconnect"/>
    <addaction name="actionQuadroReset"/>
    <addaction name="quadro_reconnect"/>
   </widget>
   <widget class="QMenu" name="menuJoystick">
    <property name="title">
     <string>Joystick</string>
    </property>
    <addaction name="actionJoystickConnect"/>
    <addaction name="actionJoystickDisconnect"/>
    <addaction name="actionJoystickCalibrate_zero"/>
    <addaction name="joy_reconnect"/>
    <addaction name="JoystickUse"/>
   </widget>
   <widget class="QMenu" name="menuProgram">
    <property name="title">
     <string>Program</string>
    </property>
    <addaction name="LogSave_data"/>
   </widget>
   <addaction name="menuProgram"/>
   <addaction name="menuQuadro"/>
   <addaction name="menuJoystick"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionQuadroConnect">
   <property name="text">
    <string>Connect</string>
   </property>
   <property name="softKeyRole">
    <enum>QAction::NegativeSoftKey</enum>
   </property>
  </action>
  <action name="actionQuadroDisconnect">
   <property name="text">
    <string>Disconnect</string>
   </property>
  </action>
  <action name="actionJoystickConnect">
   <property name="text">
    <string>Connect</string>
   </property>
  </action>
  <action name="actionJoystickDisconnect">
   <property name="text">
    <string>Disconnect</string>
   </property>
  </action>
  <action name="actionQuadroReset">
   <property name="text">
    <string>Reset</string>
   </property>
  </action>
  <action name="actionJoystickCalibrate_zero">
   <property name="text">
    <string>Calibrate zero</string>
   </property>
  </action>
  <action name="quadro_reconnect">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Reconnect</string>
   </property>
  </action>
  <action name="joy_reconnect">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Reconnect</string>
   </property>
  </action>
  <action name="LogSave_data">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Save data</string>
   </property>
  </action>
  <action name="JoystickUse">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Use</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QwtPlot</class>
   <extends>QFrame</extends>
   <header>qwt_plot.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
